staging/cse/windows/azurecnifunc.ps1

```diff
@@ -416,6 +417,22 @@ function New-ExternalHnsNetwork
         {
             $managementIP = $netIP.IPAddress
             $adapterName = $na.Name
+
+            Write-Log "Get node IPv4 address assigned to the adapter $($na.Name): $($managementIP)"
+            $nodeIPs += $managementIP
+
+            if ($IsDualStackEnabled) {
+                $netIPv6s = Get-NetIPAddress -ifIndex $na.ifIndex -AddressFamily IPv6 -ErrorAction SilentlyContinue -ErrorVariable netIPErr
+                foreach($ipv6 in $netIPv6s)
+                {
+                    if ($ipv6.PrefixOrigin -ne "WellKnown")

```

Comment:
why well known?  add a comment why this is your filter,

Reply:
ipv6 is also just taking the first nic with an ip right?

Reply:
On an Azure Windows VM, there are two IPv6 IP addresses. Below is an example. It is same in an Azure Linux VM.
```
ifIndex IPAddress                                       PrefixLength PrefixOrigin SuffixOrigin AddressState PolicyStore
------- ---------                                       ------------ ------------ ------------ ------------ -----------
6       fe80::97bd:baf7:2853:f73d%6                               64 WellKnown    Link         Preferred    ActiveStore
6       2404:f800:8000:122::4                                    128 Dhcp         Dhcp         Preferred    ActiveStore
```
From the found docuements. `fe80:` with `WellKnown` is the link-local address so we should ignore it.

IPv6 link-local is a special type of unicast address that is auto-configured on any interface using a combination of the link-local prefix FE80::/10 (first 10 bits equal to 1111 1110 10) and the MAC address of the interface. 

https://learn.microsoft.com/en-us/dotnet/api/system.net.networkinformation.prefixorigin?view=net-8.0
<html>
<body>
<!--StartFragment-->
WellKnown | 2 | The prefix is a well-known prefix. Well-known prefixes are specified in standard-track Request for Comments (RFC) documents and assigned by the Internet Assigned Numbers Authority (Iana) or an address registry. Such prefixes are reserved for special purposes.
-- | -- | --


<!--EndFragment-->
</body>
</html>

Reply:
Added the comment for the code.